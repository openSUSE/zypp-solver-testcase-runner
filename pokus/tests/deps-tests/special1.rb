#!/usr/bin/ruby


# Check handling of very special provides.


require 'zypptools/lib/package'
require 'zypptools/output/output'
include ZYppTools

require '../../src/testsuite'


packages = Array.new()

pkg = Package.new("test")
pkg.arch = "i586"
pkg.version = "1.2"
pkg.release = 1
pkg.summary = "A pure test package"
pkg.deps :provides, "hello < 2.0"
pkg.deps :provides, "world >= 2.0"
pkg.deps :provides, "modalias(kernel-default:pci:v00008086d00002580sv*sd*bc*sc*i*)"
pkg.deps :provides, "locale(aspell:csb)"
pkg.deps :provides, "perl(Automake::Options) = 0.31"
pkg.deps :provides, "xorg-x11-libs:/usr/X11R6/lib/libXt.so.6"
pkg.deps :provides, "ksym(zt_dtmf_tone) = 150c20a4"
pkg.deps :supplements, "filesystem(xfs)"
packages.push(pkg)


path = Testsuite::write_repo(:yum, packages)


Testsuite.set_arch("i586")
pool = Testsuite.read_repo("file://" + path)


Testsuite::haha2(pool).each do |res|
    puts "#{res.kind_to_s} #{res.name} #{res.edition.to_s} #{res.arch.to_s}"
    puts "  Summary: #{res.summary}"
    Testsuite::dump_all_deps(res)
    puts
end

