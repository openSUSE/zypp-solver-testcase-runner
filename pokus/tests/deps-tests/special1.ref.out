package test 1.2-1 i586
  Summary: A pure test package
  Provides: hello < 2.0
  Provides: ksym(zt_dtmf_tone) == 150c20a4
  Provides: perl(Automake::Options) == 0.31
  Provides: test == 1.2-1
  Provides: world >= 2.0
  Provides: xorg-x11-libs:/usr/X11R6/lib/libXt.so.6
  Freshens: csb
  Freshens: modalias(kernel-default:pci:v00008086d00002580sv*sd*bc*sc*i*)
  Supplements: aspell
  Supplements: filesystem(xfs)
  Supplements: kernel-default

